### DISCUSSION TITLE
Creation of some Read-only columns or read-only cells

### DISCUSSION THREAD

**Discussion by siddhu741** [exceljs/exceljs#2532] [Ideas & Proposals]
First of all, Great library and a very useful one. 
Sometimes, there is a need to make some columns of the worksheets non-editable.
can anyone throw some light on whether its possible to create such columns using the existing library?

**Comment by ekochman** [comment]
As far as I've been able to determine, this is not currently possible. However, given that locking a cell is done by setting a simple flag on the cell, this shouldn't be too complex to add support for. I'd love to have this feature as well - something like worksheet->getCell(A1)->locked = true;

**Comment by ekochman** [comment]
According to PR https://github.com/guyonroche/exceljs/pull/166 it seems that this is a currently unknown attribute type - perhaps a more general update would be to allow you to set any attribute type for a cell by providing the attribute name and value.
